src/frequenz/sdk/timeseries/_moving_window.py

```diff
@@ -119,26 +132,43 @@ def __init__(
             asyncio.CancelledError: when the task gets cancelled.
         """
         assert (
-            sampling_period.total_seconds() > 0
-        ), "The sampling period should be greater than zero."
+            input_sampling_period.total_seconds() > 0
+        ), "The input sampling period should be greater than zero."
         assert (
-            sampling_period <= size
-        ), "The sampling period should be equal to or lower than the window size."
+            input_sampling_period <= size
+        ), "The input sampling period should be equal to or lower than the window size."
+
+        sampling = input_sampling_period
+        self._resampler: Resampler | None = None
+        self._avg_sender: Sender[Sample] | None = None
+        self._resampler_task: asyncio.Task[None] | None = None
+
+        if resampler_config:
+            resampling_period = timedelta(seconds=resampler_config.resampling_period_s)
+            assert (
+                resampling_period <= size
+            ), "The resampling period should be equal to or lower than the window size."
+
+            if resampling_period != input_sampling_period:
+                self._resampler = Resampler(resampler_config)
+                sampling = resampling_period
```

Comment:
I'm not sure if this is a good idea to skip the resampling in this case. What if the user wants to configure the resampler differently? Like use a different resampling function? If it passed a resampler config explicitly with the same period it might be for a reason.

I don't know, maybe it doesn't make sense to resample at all if the input sampling rate is the same as the requested resampling rate, but ignoring it silently makes some noise in my head, if I explicitly pass a config it should be used IMHO. Another option would be to raise an exception if they match.

Reply:
I think it makes sense to remove the check and I wouldn't add an assertion given that user might just want to use the same resampling period with a different resampling function as you also mentioned it